graphql-ballerina/engine.bal

```diff
@@ -51,7 +52,7 @@ isolated class Engine {
                 message: message,
                 locations: []
             };
-            return getOutputObjectFromErrorDetail(errorDetail);
+            return [getOutputObjectFromErrorDetail(errorDetail), DOCUMENT_ERROR];
```

Comment:
This should be a different scenario (If we are to treat these as different scenarios).

There are two ways:
1. Treat all the errors that occurred before the execution as the same, (`VALIDATION_ERROR` maybe?). Then we can add that to all the places. Because we only need this information to decide whether to send `400` or `200` as the status code, I guess this approach is okay. In that case, we can even return a boolean value as the second element in the tuple. (`isValidationError` maybe?)
2. Treat different errors that occurred at different validation phases as different errors. AFAIU, we don't need that information (at least for now) to decide the status code for the response. But if we are to treat them as different errors, this is a separate error type. (`INVALID_REQUEST` maybe?)

WDYT?